src/main/java/manufacturing/dao/jdbc/CarDaoJdbcImpl.java

```diff
@@ -0,0 +1,215 @@
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't create car "
+                    + car, e);
+        }
+        return car;
+    }
+
+    @Override
+    public Optional<Car> getById(Long carId) {
+        String query = "SELECT c.car_id, c.car_model, c.manufacturer_id, "
+                + "m.manufacturer_name, m.manufacturer_country "
+                + "FROM cars c "
+                + "INNER JOIN manufacturers m "
+                + "ON m.manufacturer_id = c.manufacturer_id "
+                + "WHERE car_id = ? AND c.deleted = FALSE";
+        Car car = null;
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            preparedStatement.setLong(1, carId);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                car = createCar(resultSet, connection);
+                car.setDrivers(getAllDriversByCarId(carId, connection));
+            }
+            return Optional.ofNullable(car);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get Car by id "
+                    + carId, e);
+        }
+    }
+
+    @Override
+    public Car update(Car car) {
+        String deletionQuery = "DELETE FROM cars_drivers WHERE car_id = ?";
+        String updateQuery = "UPDATE cars "
+                + "SET car_model = ?, manufacturer_id = ? "
+                + "WHERE car_id = ? AND deleted = FALSE";
+        String insertQuery = "INSERT INTO cars_drivers "
+                + "(car_id, driver_id) VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatementForDeletion =
+                        connection.prepareStatement(deletionQuery);
+                PreparedStatement preparedStatementForUpdating =
+                        connection.prepareStatement(updateQuery);
+                PreparedStatement preparedStatementForInserting =
+                        connection.prepareStatement(insertQuery)) {
+            preparedStatementForDeletion.setLong(1, car.getId());
+            preparedStatementForDeletion.executeUpdate();
+            preparedStatementForUpdating.setString(1, car.getModel());
+            preparedStatementForUpdating.setLong(2, car.getManufacturer().getId());
+            preparedStatementForUpdating.setLong(3, car.getId());
+            preparedStatementForUpdating.executeUpdate();
+            preparedStatementForInserting.setLong(1, car.getId());
+            for (Driver driver : car.getDrivers()) {
+                preparedStatementForInserting.setLong(2, driver.getId());
+                preparedStatementForInserting.executeUpdate();
+            }
+            return car;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update car in DB", e);
+        }
```

Comment:
```suggestion
        String updateQuery = "UPDATE cars "
                + "SET car_model = ?, manufacturer_id = ? "
                + "WHERE car_id = ? AND deleted = FALSE";
        try (Connection connection = ConnectionUtil.getConnection();
                PreparedStatement preparedStatementForUpdating =
                        connection.prepareStatement(updateQuery)) {
            preparedStatementForUpdating.setString(1, car.getModel());
            preparedStatementForUpdating.setLong(2, car.getManufacturer().getId());
            preparedStatementForUpdating.setLong(3, car.getId());
            preparedStatementForUpdating.executeUpdate();
        } catch (SQLException e) {
            throw new DataProcessingException("Can't update car in DB", e);
        }
        removeDriversFromCar(car);
        addDriversToCar(car);
        return car;
```
let's divide the logic into several methods, each opening its own connection
we will be able to reuse these methods if we want

Reply:
done as well